static/js/components/NicDeviceSelector.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useState,
+} from "react";
+import { Input, Select } from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { fetchNetworks } from "../api/networks";
+import { NotificationHelper } from "../types/notification";
+import { queryKeys } from "../util/queryKeys";
+import { LxdNicDevice } from "../types/device";
+
+interface Props {
+  notify: NotificationHelper;
+  setNicDevice: Dispatch<SetStateAction<LxdNicDevice | null>>;
+}
+
+const NicDeviceSelector: FC<Props> = ({ notify, setNicDevice }) => {
+  const [nic, setNic] = useState<string>("");
+  const [nameInInstance, setNameInInstance] = useState<string>("");
+
+  const { data: networks = [], error } = useQuery({
+    queryKey: [queryKeys.networks],
+    queryFn: fetchNetworks,
+  });
+
+  if (error) {
+    notify.failure("Could not load networks.", error);
+  }
+
+  const getNetworkOptions = () => {
+    const options = networks.map((network) => {
+      return {
+        label: network.name,
+        value: network.name,
+        disabled: false,
+      };
+    });
+    options.unshift({
+      label: networks.length === 0 ? "No network available" : "Select option",
+      value: "",
+      disabled: true,
+    });
+    return options;
+  };
+
+  const updateNic = (v: ChangeEvent<HTMLSelectElement>) => {
+    setNic(v.target.value);
+    setNicDevice({
+      name: nameInInstance,
+      network: v.target.value,
+      type: "nic",
+    });
+  };
+
+  const updateNameInInstance = (v: ChangeEvent<HTMLInputElement>) => {
+    setNameInInstance(v.target.value);
+    setNicDevice({
+      name: v.target.value,
+      network: nic,
+      type: "nic",
+    });
+  };
+
+  return (
+    <>
+      <h4>Network device</h4>
+      <Select
+        name="nic"
+        label="Select a network device (NIC)"
+        options={getNetworkOptions()}
+        onChange={updateNic}
+        value={nic}
+        stacked
+      />
+      <Input
+        id="nameInInstance"
```

Comment:
Shall we add a placeholder with a hint what possible values look like? I.e. eth0 or what is commonly used